shared_model/backend/protobuf/proto_tx_status_factory.hpp

```diff
@@ -0,0 +1,68 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef IROHA_PROTO_TX_STATUS_FACTORY_HPP
+#define IROHA_PROTO_TX_STATUS_FACTORY_HPP
+
+#include "backend/protobuf/transaction_responses/proto_tx_response.hpp"
+#include "interfaces/iroha_internal/tx_status_factory.hpp"
+
+namespace shared_model {
+  namespace proto {
+    class ProtoTxStatusFactory : public interface::TxStatusFactory {
+     private:
+      using FactoryReturnType = interface::TxStatusFactory::FactoryReturnType;
+
+     public:
+      // ------------------------| Stateless statuses |-------------------------
+
+      FactoryReturnType makeTxStatusStatelessFail(TransactionHashType,
+                                                  ErrorMessageType) override;
+
+      FactoryReturnType makeTxStatusStatelessValid(TransactionHashType,
+                                                   ErrorMessageType) override;
+
+      // ------------------------| Stateful statuses |--------------------------
+
+      FactoryReturnType makeTxStatusStatefulFail(TransactionHashType,
+                                                 ErrorMessageType) override;
+      FactoryReturnType makeTxStatusStatefulValid(TransactionHashType,
+                                                  ErrorMessageType) override;
+
+      // ---------------------------| End statuses |----------------------------
+
+      FactoryReturnType makeTxStatusCommitted(TransactionHashType,
+                                              ErrorMessageType) override;
+
+      FactoryReturnType makeTxStatusRejected(TransactionHashType,
+                                             ErrorMessageType) override;
+
+      // --------------------------| Rest statuses |----------------------------
+
+      FactoryReturnType makeTxStatusMstExpired(TransactionHashType,
+                                               ErrorMessageType) override;
+
+      FactoryReturnType makeTxStatusNotReceived(TransactionHashType,
+                                                ErrorMessageType) override;
+
+      FactoryReturnType makeEnoughSignaturesCollected(
+          TransactionHashType, ErrorMessageType) override;
+
+     private:
+      /**
+       * Fills common fields for all statuses
+       */
+      iroha::protocol::ToriiResponse fillCommon(TransactionHashType,
```

Comment:
That function and the following can be non-class function. So if internal implementation changed, there's no need in all dependency recompilation. Anonymous namespaces can be helpful